src/Compilers/Test/Core/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -824,8 +830,11 @@ PooledHashSet<CaptureId> adjustAndGetEntryState(ArrayBuilder<PooledHashSet<Captu
                                      isIncrementedNullableForToLoopControlVariable(reference) ||
                                      isConditionalAccessReceiver(reference) ||
                                      isCoalesceAssignmentTarget(reference) ||
-                                     isObjectInitializerInitializedObjectTarget(reference)) &&
-                                 block.EnclosingRegion.EnclosingRegion.CaptureIds.Contains(id)),
+                                     isObjectInitializerInitializedObjectTarget(reference) ||
+                                     isInterpolatedStringArgumentCapture(reference) ||
+                                     isInterpolatedStringHandlerCapture(reference)) &&
+                                 (block.EnclosingRegion.EnclosingRegion.CaptureIds.Contains(id) ||
```

Comment:
>block.EnclosingRegion.EnclosingRegion.CaptureIds.Contains(id) [](http://example.com/codeflow?start=34&length=61)

Would it be possible to bundle this with ```isObjectInitializerInitializedObjectTarget(reference)``` (as it used to be). And bundle ```(block.EnclosingRegion.EnclosingRegion?.EnclosingRegion.CaptureIds.Contains(id) ?? false)``` with interpolation captures. The goal is to make the check stronger.

Reply:
Yes, but I'm going to break this condition up to do so. It's at my complexity limit now, and trying to nest more things breaks my brain reading it.

Reply:
Actually, I'm not sure what you mean by the first part. If I move the `block.EnclosingRegion.EnclosingRegion.CaptureIds.Contains(id)` check, it weakens the rest of the checks here by allowing them to exist anywhere in the graph.

Reply:
I did my best interpretation of what I think you meant. Let me know if it's not what you had in mind.

Reply:
> I did my best interpretation of what I think you meant. Let me know if it's not what you had in mind.

Sorry for the confusion. I think I slightly misread the original condition, but you adapted the suggestion correctly.